source/client.c

```diff
@@ -491,12 +486,18 @@ int aws_mqtt_client_connection_connect(
 
     connection->host_name = aws_string_new_from_array(connection->allocator, host_name->ptr, host_name->len);
     connection->port = port;
-    connection->tls_options = tls_options;
-    connection->socket_options = socket_options;
+    connection->socket_options = *socket_options;
     connection->state = AWS_MQTT_CLIENT_STATE_CONNECTING;
     connection->clean_session = clean_session;
     connection->keep_alive_time = keep_alive_time;
 
+    /* Cheat and set the tls_options host_name to our copy if they're the same */
+    if (tls_options) {
```

Comment:
Since connect() can theoretically be called multiple times, should we zero out connection->tls_options if NULL was passed in?

Reply:
Yes!